From aa3eef707ae81cd8bab0d7eb94a8729c59061f61 Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Thu, 28 Nov 2024 14:58:12 +0100
Subject: [PATCH] nfs: Set intitial_offset when appending

Currently the `initial_offset` is not set when appening. This is wrong
as the `GDaemonOutputStream` implemetation relies on it to be set. Let's
do so to fix the `g_seekable_tell` output.
---
 daemon/gvfsbackendnfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/daemon/gvfsbackendnfs.c b/daemon/gvfsbackendnfs.c
index 6219895a..85632174 100644
--- a/daemon/gvfsbackendnfs.c
+++ b/daemon/gvfsbackendnfs.c
@@ -759,11 +759,10 @@ append_stat_cb (int err,
   GVfsJob *job = G_VFS_JOB (private_data);
   GVfsJobOpenForWrite *op_job = G_VFS_JOB_OPEN_FOR_WRITE (job);
   GVfsBackendNfs *op_backend = G_VFS_BACKEND_NFS (op_job->backend);
+  struct nfs_stat_64 *st = data;
 
   if (err == 0)
     {
-      struct nfs_stat_64 *st = data;
-
       if (S_ISDIR (st->nfs_mode))
         {
           g_vfs_job_failed_literal (job,
@@ -778,6 +777,7 @@ append_stat_cb (int err,
       return;
     }
 
+  g_vfs_job_open_for_write_set_initial_offset (op_job, st->nfs_size);
   nfs_create_async (op_backend->ctx,
                     op_job->filename,
                     O_APPEND,
-- 
2.46.2

